src/CommaSeparatedParser.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+namespace MPScholten\RequestParser;
+
+class CommaSeparatedParser extends AbstractValueParser
+{
+    const INT = 'INT';
+    const STRING = 'STRING';
+    const FLOAT = 'FLOAT';
+    const BOOLEAN = 'BOOLEAN';
+    const YES_NO_BOOLEAN = 'YES_NO_BOOLEAN';
+    const JSON = 'JSON';
+    const DATE_TIME = 'DATE_TIME';
+
+    private $type;
+
+    protected function parse($value)
+    {
+        if (!isset($this->type)) {
+            $this->type = self::STRING;
+        }
+        $valuesArray = explode(",", $value);
+        switch ($this->type) {
```

Comment:
Good point, I also think that there's a general problem with this solution's flow, but I'll pick up from where I stopped and will think of something more constructive :)